packages/pigeon/tool/generate.dart

```diff
@@ -19,12 +19,21 @@ import 'shared/generation.dart';
 
 const String _helpFlag = 'help';
 const String _formatFlag = 'format';
+const String _files = 'files';
+const String _test = 'test';
+const String _example = 'example';
+
+const List<String> _fileGroups = <String>[_test, _example];
 
 Future<void> main(List<String> args) async {
   final ArgParser parser = ArgParser()
     ..addFlag(_formatFlag, abbr: 'f', help: 'Autoformats after generation.')
     ..addFlag(_helpFlag,
-        negatable: false, abbr: 'h', help: 'Print this reference.');
+        negatable: false, abbr: 'h', help: 'Print this reference.')
+    ..addOption(_files,
```

Comment:
Optional: you could future-proof this against more options by making it a multi-option, and checking for inclusion instead of equality later, so that if we had groups A, B, and C at some point you could do `--files=A,C`.

Reply:
good idea, didn't realize there was an option for that.